bin/pycbc_live

```diff
@@ -985,20 +988,45 @@ with ctx:
                 results.pop(ifo)
 
             # Veto single detector triggers that fail the DQ vector
+            # or have iDQ value below the threshold
             for ifo in results:
-                if data_reader[ifo].dq is None:
-                    continue
-                logging.info("Checking %s's DQ vector", ifo)
-                start = data_reader[ifo].start_time
-                times = results[ifo]['end_time']
-                idx = data_reader[ifo].dq.indices_of_flag(
-                        start, valid_pad, times,
-                        padding=data_reader[ifo].dq_padding)
-                logging.info('Keeping %d/%d %s triggers',
-                             len(idx), len(times), ifo)
-                for key in results[ifo]:
-                    if len(results[ifo][key]):
-                        results[ifo][key] = results[ifo][key][idx]
+                if data_reader[ifo].dq is not None:
+                    logging.info("Checking %s's DQ vector", ifo)
+                    start = data_reader[ifo].start_time
+                    times = results[ifo]['end_time']
+                    idx = data_reader[ifo].dq.indices_of_flag(
+                            start, valid_pad, times,
+                            padding=data_reader[ifo].dq_padding)
+                    logging.info('Keeping %d/%d %s triggers',
+                                 len(idx), len(times), ifo)
+                    for key in results[ifo]:
+                        if len(results[ifo][key]):
+                            results[ifo][key] = results[ifo][key][idx]
+                if data_reader[ifo].idq is not None:
+                    logging.info("Checking %s's iDQ information", ifo)
+                    start = data_reader[ifo].start_time
+                    times = results[ifo]['end_time']
+                    # if idq_states = 0, idq info was invalid. Don't let
+                    # those triggers get vetoed
+                    idq_states = data_reader[ifo].idq_state.at_times(times)
+                    idq_valid = idq_states == 1
+                    idq_invalid = numpy.logical_not(idq_valid)
+                    invalid_idx = numpy.where(idq_invalid)[0]
+
+                    # for triggers with valid idq info, only keep those with
+                    # idq FAP > threshold
+                    valid_idx = numpy.where(idq_valid)[0]
+                    valid_keep = data_reader[ifo].idq.indices_of_flag(
```

Comment:
I am a bit confused by what is being done here. What is the reasoning behind using `at_times()` for the iDQ status channel and `indices_of_flag()` for the iDQ FAP channel? Where are you actually checking that the iDQ FAP is above the given threshold?

Reply:
Sorry I overlooked the code that answers the second question, but the first one stands.